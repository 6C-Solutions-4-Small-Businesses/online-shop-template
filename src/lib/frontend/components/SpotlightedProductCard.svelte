<script lang="ts">
    import ShoppingCartActions from '$lib/frontend/components/ShoppingCartActions.svelte'
    import Image from '$lib/frontend/components/Image.svelte'

    export let id: string
    export let image: string
    export let name: string
    export let price: number
    export let salePrice: number | undefined | null = null

    let isProductHovered = false
</script>

<div class="w-full relative flex flex-col bg-white"
     on:focus={() => isProductHovered = true}
     on:mouseleave={()=> isProductHovered = false}
     on:mouseover={()=> isProductHovered = true}
     role="none"
>
    <Image imageRemoteUrl="{image}" name={name}
           classes="object-contain h-[15rem] sm:h-[30rem] md:h-[30rem] lg:h-[30rem] xl:h-[30rem] py-5"
            width="{640}"/>
    <div
            class={`h-16 flex items-center justify-center gap-2 text-3xl rounded-none ${isProductHovered ? 'bg-green-50' : 'bg-white'}`}
    >
        <span>{name}</span>
        <span class="font-bold text-red-500"> {price / 100 >= 1 ? "$" : "Â¢"}{price / 100}</span>
    </div>
    <div class={`${isProductHovered ? 'w-full md:w-2/3 absolute top-0 right-0 flex justify-end' : 'hidden'}`}>
        <ShoppingCartActions
                image={image}
                name={name}
                productId={id}
                regularPrice={price}
                salePrice={salePrice}
        />
    </div>
</div>
