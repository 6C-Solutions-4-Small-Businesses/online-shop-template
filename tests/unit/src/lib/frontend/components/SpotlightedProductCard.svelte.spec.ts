import {describe, expect} from 'vitest'
import {render} from '@testing-library/svelte'
import SpotlightedProductCard from '$lib/frontend/components/SpotlightedProductCard.svelte'

describe('Spotlight product card', () => {
    let container: HTMLElement

    beforeEach(() => {
        ({container} = render(SpotlightedProductCard, {
            props: {
                id: 'product-id',
                name: 'product-name',
                image: 'product-image',
                regularPrice: 24.99,
                isSoldByQuantities: false,
                unit: 'unit',
                salePrice: 19.99,
            }
        }))
    })

    it('should have match snapshot', () => {
        expect(container).toMatchSnapshot()
    })
})