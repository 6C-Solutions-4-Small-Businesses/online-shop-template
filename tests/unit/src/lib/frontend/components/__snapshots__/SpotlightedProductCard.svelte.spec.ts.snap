// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Spotlight product card > should have match snapshot 1`] = `
<body>
  <div>
    <div
      class="w-full relative flex flex-col bg-white"
      role="none"
    >
      <img
        alt="product-name"
        class="object-contain h-[15rem] sm:h-[30rem] md:h-[30rem] lg:h-[30rem] xl:h-[30rem] py-5"
        src="product-image"
        width="640"
      />
      <!--&lt;Image&gt;-->
       
      <div
        class="h-16 flex items-center justify-center gap-2 text-3xl rounded-none bg-white"
      >
        <span>
          product-name
        </span>
         
        <span
          class="font-bold text-red-500"
        >
          Â¢
          0.24989999999999998
        </span>
      </div>
       
      <div
        class="hidden"
      >
        <div
          class="text-2xl  w-2/12 h-10 flex items-center justify-center text-red-500"
          role="region"
        >
           
          <div
            class="border border-red-500 w-full h-full flex justify-center items-center bg-red-100"
          >
            <svg
              height="1.2em"
              viewBox="0 0 24 24"
              width="1.2em"
            >
              <path
                d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3M7 18c-1.1 0-2 .9-2 2s.9 2 2 2s2-.9 2-2s-.9-2-2-2m10 0c-1.1 0-2 .9-2 2s.9 2 2 2s2-.9 2-2s-.9-2-2-2m-9.8-3.2v-.1l.9-1.7h7.4c.7 0 1.4-.4 1.7-1l3.9-7l-1.7-1l-3.9 7h-7L4.3 2H1v2h2l3.6 7.6L5.2 14c-.1.3-.2.6-.2 1c0 1.1.9 2 2 2h12v-2H7.4c-.1 0-.2-.1-.2-.2"
                fill="currentColor"
              />
            </svg>
            <!--&lt;Cart-plus&gt;-->
          </div>
        </div>
        <!--&lt;ShoppingCartActions&gt;-->
      </div>
    </div>
    <!--&lt;SpotlightedProductCard&gt;-->
  </div>
</body>
`;
